activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/behavior/MQServiceTaskBehavior.java

```diff
@@ -90,8 +91,16 @@ private void enrichIntegrationContext(IntegrationContext integrationContext) {
 
     private void aggregateCloudIntegrationRequestedEvent(IntegrationContext integrationContext) {
         if (runtimeBundleProperties.getEventsProperties().isIntegrationAuditEventsEnabled()) {
-            CloudIntegrationRequestedEventImpl cloudEvent = new CloudIntegrationRequestedEventImpl(integrationContext);
-
+            CloudIntegrationRequestedEventImpl cloudEvent;
+            if (integrationContext.hasEphemeralVariables()) {
+                IntegrationContext sanitizedContext = new IntegrationContextImpl(
+                    (IntegrationContextImpl) integrationContext
```

Comment:
It's preferable to avoid casts as much as possible, so please, have a look in the related constructor to see if it's possible to pass the interface rather than the implementation.